### PR TITLE
fiberzerolog: Use string for latency field

### PR DESCRIPTION
**Pull request by raphielscape**
Fixes https://github.com/gofiber/contrib/issues/704

### CODE REVIEW

**Comment by ReneWerner87** [comment]
Pls provide a documentation change and unittests for what this change involves so that it remains audited and documented

**Comment by gaby** [comment]
@ReneWerner87 The current tests dont check any of the fields used by this middleware

**Comment by gaby** [comment]
Actually it does check fields, but doesnt check latency at all. We can at least add a test to check the field type.

The regular logger and fiberzap all return "Strings" for latency but this middleware is not. The changes are more of a consistency thing.

**Comment by gaby** [comment]
`govulncheck` failure is related to go1.20.7 being released. Related: https://github.com/actions/setup-go/issues/405

**Comment by raphielscape** [comment]
I have added the unit test for latency

**Comment by gaby** [comment]
@raphielscape Thanks! Could you also added in the test at the bottom for "Logger All", the one with all the fields?

**Comment by gaby** [comment]
This test: https://github.com/gofiber/contrib/blob/6fa822cca2e7421e5a844746068ea4986f2f8d5c/fiberzerolog/zerolog_test.go#L143

**Comment by raphielscape** [comment]
Done

**Comment by gaby** [comment]
@raphielscape What I was thinking is adding the latency field on this list here: https://github.com/gofiber/contrib/blob/6fa822cca2e7421e5a844746068ea4986f2f8d5c/fiberzerolog/zerolog_test.go#L152 and then checking the expected value below. 

You added a whole new Fiber App in that test which already has a FiberApp

**Comment by raphielscape** [comment]
> @raphielscape What I was thinking is adding the latency field on this list here:
> 
> https://github.com/gofiber/contrib/blob/6fa822cca2e7421e5a844746068ea4986f2f8d5c/fiberzerolog/zerolog_test.go#L152
> 
> and then checking the expected value below.
> You added a whole new Fiber App in that test which already has a FiberApp

I have explained it in the commit description, we can't use the same app because we can't do Equal Assertion as latency varies between runs

**Comment by raphielscape** [comment]
We can make the first fiber app have a endpoint that returns OK and use that in test instead, though

**Review by gaby — APPROVED** [review]
LGTM

**Comment by gaby** [comment]
> @gaby good?

Thanks! That looks good

File: fiberzerolog/zerolog_test.go [hunk 1]
```diff
@@ -181,15 +222,25 @@ func Test_Logger_All(t *testing.T) {
...
 	_ = json.Unmarshal(buf.Bytes(), &logs)
 
-	utils.AssertEqual(t, expected, logs)
+	for key, value := range expected {
+		utils.AssertEqual(t, value, logs[key])
+	}
+
+	latencyStr, ok := logs[FieldLatency].(string)
+	if !ok {
+		t.Errorf("Failed to parse latency from logs")
```
Comment (efectn) [inline fiberzerolog/zerolog_test.go:239]: use utils.AssertEqual instead

File: fiberzerolog/zerolog_test.go [hunk 2]
```diff
@@ -181,15 +222,25 @@ func Test_Logger_All(t *testing.T) {
...
+	}
+	if !strings.Contains(latencyStr, "ms") {
+		t.Errorf("Latency does not contain 'ms': %s", latencyStr)
```
Comment (efectn) [inline fiberzerolog/zerolog_test.go:242]: same

File: fiberzerolog/zerolog_test.go [hunk 3]
```diff
@@ -125,6 +127,39 @@ func Test_Logger(t *testing.T) {
...
+	resp, err := app.Test(httptest.NewRequest("GET", "/", nil))
+	utils.AssertEqual(t, nil, err)
+	utils.AssertEqual(t, fiber.StatusOK, resp.StatusCode)
+
+	var logs map[string]any
+	_ = json.Unmarshal(buf.Bytes(), &logs)
+
+	latencyStr, ok := logs[FieldLatency].(string)
+	if !ok {
+		t.Errorf("Failed to parse latency from logs")
```
Comment (efectn) [inline fiberzerolog/zerolog_test.go:155]: here too

File: fiberzerolog/zerolog_test.go (hunk 1 above)
Comment (raphielscape) [reply fiberzerolog/zerolog_test.go:239]: We can't use AssertEqual because latency always varies and we can't have an equal value for AssertEqual

Comment (gaby) [reply fiberzerolog/zerolog_test.go:239]: @efectn We can only check the type for this field.

**Comment by raphielscape** [comment]
For the reason why I don't use AssertEqual is that I have no idea how to make an equally valued latency and latency return always varies between test runs, please let me know if there was a way to make the latency in an equilibrium so that we got an equal value that can be compared so that I can use AssertEqual

File: fiberzerolog/zerolog_test.go (hunk 1 above)
Comment (efectn) [reply fiberzerolog/zerolog_test.go:239]: > @efectn We can only check the type for this field.

Isn't ok boolean? We can check it's true

Comment (raphielscape) [reply fiberzerolog/zerolog_test.go:239]: OK came from interface type assertion, not the latency return itself, we can't AssertEqual that

Comment (efectn) [reply fiberzerolog/zerolog_test.go:239]: > OK came from interface type assertion, not the latency return itself, we can't AssertEqual that

I know, i just mean something like
```go
utils.AssertEqual(t, true, ok)
utils.AssertEqual(t, true, strings.Contains(latencyStr, "ms"))
```

Comment (raphielscape) [reply fiberzerolog/zerolog_test.go]: > > OK came from interface type assertion, not the latency return itself, we can't AssertEqual that
> 
> I know, i just mean something like
> 
> ```go
> utils.AssertEqual(t, true, ok)
> utils.AssertEqual(t, true, strings.Contains(latencyStr, "ms"))
> ```

Okay, that works

**Review by efectn — APPROVED** [review]

**Pull request merged**